modeltrans/utils.py

```diff
@@ -25,4 +28,76 @@ def build_localized_fieldname(field_name, lang):
         # The 2-letter Indonesian language code is problematic with the
         # current naming scheme as Django foreign keys also add "id" suffix.
         lang = "ind"
-    return str("{}_{}".format(field_name, lang.replace("-", "_")))
+    return "{}_{}".format(field_name, lang.replace("-", "_"))
+
+
+def get_model_field(model, path):
+    """
+    Return the django model field for model in context, following relations.
+    """
+    if not hasattr(model, "_meta"):
+        return
+
+    field = None
+    for bit in path.split(LOOKUP_SEP):
+        try:
+            field = model._meta.get_field(bit)
+        except FieldDoesNotExist:
+            break
+
+        if hasattr(field, "remote_field"):
+            rel = getattr(field, "remote_field", None)
+            model = getattr(rel, "model", model)
```

Comment:
Begrijp ik dit goed als:
- "remote_field" is een standaard Model attr die een foreignkey aangeeft?
- als die bestaat dan vraag je remote field met naam "field" op? "rel" is hier "relationship"?
- en dan haal je het desbetreffende model instance op?
Nee dat lijkt me niet correct. Enfin, ik volg het niet helemaal goed, maar ik interpreteer dat je hier door schuift naar het volgende model instance (via de foreignkey) in de keten van model instances.

Reply:
Deze code is een versimpeling van https://github.com/jieter/django-tables2/blob/60d0bb5bbb1568f682eac21e9143f1643ddad64c/django_tables2/utils.py#L287-L450

Ik heb dus niet alles helemaal paraat, maar het gaat hier allemaal over niet-instances van models. Je wilt vragen beantwoorden als 'heeft model X het veld Y' of 'heeft model X een foreign key naar Y en heeft Y een foreign key naar Z en heeft Z een veld A'.